### DISCUSSION TITLE
MQTT unsubscribe

### DISCUSSION THREAD

**Discussion by lonejack** [cesanta/mongoose#2662] [Ask questions, get answers] [answered]
Hi, what is the canonical API to unsubscribe?
On doc:

> https://mongoose.ws/documentation/#mqtt-1

seems not present...

**Comment by scaprile** [comment, marked as answer]
No, we don't have one. Mongoose is not a generic client like mosquitto_sub, embedded systems usually subscribe to their corresponding topics and stay subscribed.
You can check the broker example/tutorial for a grasp on a lower level and write your own if you need it.
https://mongoose.ws/documentation/tutorials/mqtt-server/

**Comment by scaprile** [comment]
#3407
